published_date/published_date.py

```diff
@@ -0,0 +1,109 @@
+import logging, os
+import time
+import json
+from json.decoder import JSONDecodeError
+from requests_futures.sessions import FuturesSession
+from concurrent.futures import as_completed
+
+
+logger = logging.getLogger(__name__)
+
+
+class FetchPublishedDate:
+
+    def __init__(self, canvas_url, canvas_token, workers, canvas_ids):
+        self.canvas_url = canvas_url,
+        self.canvas_token = canvas_token,
+        self.canvas_ids = canvas_ids
```

Comment:
May want to call these `course_ids` or `canvas_course_ids` for clarity.

Reply:
I just synced with naming conventions used in the `slim_down_course_data`

Reply:
I'm just concerned that this class makes no reference to a course entity in `__init__.py` or its name. Not a huge deal in the short term.